### ISSUE TITLE
CheatEngine Option Broken

### ISSUE THREAD

**Issue by zany130** [sonic2kk/steamtinkerlaunch#380]
[Trails of Cold Steel.log](https://github.com/frostworx/steamtinkerlaunch/files/7730124/Trails.of.Cold.Steel.log)
First CheatEngine fails to download if the exe is  not found in the config directory (from my understanding it should get it from the waybackmachine but the download fails for some reason) second even if you get it downloaded and installed (by manualing downloading and placing exe into the `downloads/cheatengine` folder it will open cheat engine but not the game. closing cheat engine brings the game up, but there is no obvious way to get cheat engine back up (maybe an option in the steamtinkerlauncher tray icon?) Also exiting the game apparently doesn't correctly exit everything as steam complains it is still running after close ( This does not happen if cheatengine is not enabled)

**Comment by frostworx** [comment]
This has low priority for me, because I do not care about CE at all and haven't touched it
since they removed the 7.2 binaries from their github repo.
Leaving this open, maybe someone interested is willing to look into this.

**Comment by frostworx** [comment]
Just a quick update:
The download from waybackmachine was recently contributed and I did not test it because of above reasons.
wayback machine often is slow, so maybe the download timeouted(?)
CE should be started forked, means in parallel to the game. This seems to work looking at your log:
`Thu Dec 16 02:58:24 PM EST 2021 INFO - extProtonRun - Starting '/home/zany130/.config/steamtinkerlaunch/cheatengine/Cheat Engine 7.2/Cheat Engine.exe' forked into the background`

You probably mean that CE is still running while the game is running, but it is simply "behind" the game, right?
IIRC this was always like this and you simply have to alt+tab it to the front.
I rewrote game exiting once, because it was broken for several proton versions, no idea if this might be related.

**Comment by frostworx** [comment]
Any attempt to help with feature requests and even if it is answering questions is always welcome of course.
also fyi: https://github.com/frostworx/steamtinkerlaunch/pull/360#issuecomment-997176786

**Comment by frostworx** [comment]
alt+tab stil works as it did when I added CE support _(and steamtinkerlaunch does not affect that anyway in either direction)_
assuming invalid, but please don't hesitate to elaborate or create a PR with a "fix"

**Comment by zany130** [comment]
@frostworx (pinging as I don't know if you would get notified on a closed issue) figured out what the issue was. when you have both cheat engine and gamemode enabled cheat engine is opened but not the game. no clue why it does this though. also I can still reproduce the issue of having cheat engine enabled doesn't allow the game to cleanly exit on steam

**Comment by frostworx** [comment]
to be committed _soon_
```
				if [ "$USEGAMEMODERUN" -eq 1 ]; then
					writelog "WARN" "${FUNCNAME[0]} - Disabling '$GMR' for '$CE', because function 'GAMEPID' doesn't find the game pid with it in the current state"
					USEGAMEMODERUN=0
				fi
```
